controller/cache/scan.go

```diff
@@ -619,7 +619,13 @@ func scannerDBChange(newVer string) {
 	go func() {
 		scanMutexLock()
 		for id, info := range scanMap {
-			if info.status == statusScanNone && info.version != newVer {
+			if info.version == newVer { // no need for new scan
+				continue
+			}
+			if info.status == statusScanning || info.status == statusScanNone {
```

Comment:
This fix means the reported issue only occurres to workloads that are being scanned when DB version change.
Is this what customer encountered?